src/Performance.h

```diff
@@ -0,0 +1,60 @@
+#ifndef PERFORMANCE_H
+#define PERFORMANCE_H
+
+#include "Criteria.h"
+#include "Perf.h"
+#include <iostream>
+#include <string>
+#include <vector>
+
+class Performance {
+public:
+  /**
+   * Performance standard constructor
+   *
+   * @param id Id name of the performance (ex: name of alternative or profile)
+   * @param criteria Array of criterion to compute performance over
+   */
+  Performance(std::string id, Criteria &criteria);
+
+  /**
+   * Performance standard constructor
+   *
+   * @param id Id name of the performance (ex: name of alternative or profile)
+   * @param criteria Array of criterion to compute performance over
+   * @param performance Array of performance (float) to set
+   */
+  Performance(std::string id, Criteria &criteria, std::vector<float> &p);
+
+  /**
+   * Performance standard constructor
+   *
+   * @param performance[] Array of Perf to set
+   */
+  Performance(std::vector<Perf> &p);
+
+  /**
+   * Performance constructor by copy
+   *
+   * @param perf Based performance to copy
+   */
+  Performance(const Performance &p);
+
+  ~Performance();
+
+  friend std::ostream &operator<<(std::ostream &out, const Performance &p);
+
+  /**
+   * getPerf getter of weight parameter
+   *
+   * @return perf
+   */
+  std::vector<Perf> getPerf() const;
+
```

Comment:
Not sure why we are talking about the weight parameter here.
If I'm not mistaken, the weight parameter only plays in for the Criteria object and not the performances of alternatives or profiles. What we would be taking about here, would be to get the evaluation of the performance on a given criteria.
ie if performance = [(a1,c1,0.1), (a1,c2,0.43), (a1,c3, 0.54)]
the method getPerf() would probably need an integer argument or criterion argument to indicate which value of Perf object you want.

Reply:
That's a copy paste mistake, we are talking about a std::vector<Perf> here, not a weight parameter. I'll add a [] operator (or similar) to get data from a specific criteria